### PR TITLE
pj-rehearse: implement rehearsing of changed registry steps

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds support for choosing presubmits to rehearse based on changed registry steps.

### CODE REVIEW

File: pkg/rehearse/jobs.go [hunk 1]
```diff
@@ -408,6 +408,77 @@ func AddRandomJobsForChangedTemplates(templates []config.ConfigMapSource, toBeRe
...
+	for k := range configs {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		ciopConfig := configs[key]
+		tests := ciopConfig.Configuration.Tests
+		repoPresubmits := prConfigPresubmits[ciopConfig.Info.Repo]
+		for _, test := range tests {
+			if test.MultiStageTestConfiguration != nil {
```
Comment (petr-muller) [inline pkg/rehearse/jobs.go]: I think we can make this a bit more readable by inverting this into a guard clause, allowing to de-indent the rest of the block.

File: pkg/config/release.go [hunk 2]
```diff
@@ -6,8 +6,11 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/openshift/ci-tools/pkg/registry"
```
Comment (petr-muller) [inline pkg/config/release.go]: I think this should be at the bottom (standard, third-party, local)

File: cmd/pj-rehearse/main.go [hunk 3]
```diff
@@ -185,6 +185,27 @@ func rehearseMain() int {
...
+		return gracefulExit(o.noFail, misconfigurationOutput)
+	}
+	changedRegistrySteps, err := config.GetChangedRegistrySteps(o.releaseRepoPath, jobSpec.Refs.BaseSHA, graph)
+	if err != nil {
+		logger.WithError(err).Error("could not get step registry differences")
+		return gracefulExit(o.noFail, misconfigurationOutput)
+	}
+	if len(changedRegistrySteps) != 0 {
+		logger.WithField("registry", changedRegistrySteps).Info("registry steps changed")
+		// TODO: add metrics for changed registry steps
```
Comment (stevekuznetsov) [inline cmd/pj-rehearse/main.go]: Outstanding?

File: pkg/rehearse/jobs.go [hunk 4]
```diff
@@ -408,6 +410,125 @@ func AddRandomJobsForChangedTemplates(templates []config.ConfigMapSource, toBeRe
...
+			for _, added := range addedConfigs {
+				if reflect.DeepEqual(test.MultiStageTestConfiguration, added) {
+					skip = true
+					break
+				}
+			}
+			if skip {
+				continue
+			}
+			jobName := ciopConfig.Info.JobName("pull", test.As)
```
Comment (stevekuznetsov) [inline pkg/rehearse/jobs.go]: Could we have `"pull"` as a typed constant like `presubmitPrefix` or something?

**Pull request merged**